tests/macros/InterfaceOverview.test.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * Test the API Interface Overview page content macro
+ *
+ * @prettier
+ */
+
+// Get necessary modules
+const fs = require('fs');
+const path = require('path');
+
+const { assert, describeMacro, beforeEachMacro, itMacro } = require('./utils');
+const jsdom = require('jsdom');
+const { JSDOM } = jsdom;
+
+const INTERFACE_DATA = `[${JSON.stringify({
+    XMLHttpRequestEventTarget: {
+        inh: 'EventTarget',
+        impl: []
+    }
+})}]`;
+
+const DATA_DIR = path.resolve(__dirname, 'fixtures', 'interface-overview');
+const DATA = ((...dataNames) => {
+    /** @type {Record<string, object>} */
+    let result = {};
+    for (let i = 0; i < dataNames.length; i++) {
+        const name = dataNames[i];
+        result[name] = JSON.parse(
+            fs.readFileSync(path.resolve(DATA_DIR, `${name}.json`))
+        );
+    }
+    return result;
+})('EventTarget', 'XMLHttpRequestEventTarget');
+
+/**
+ * @param {string[]} headers
+ * @returns {(result: string) => void}
+ */
+function testInterface(headers) {
+    return result => {
+        const dom = JSDOM.fragment(result);
+        expect(
+            Array.from(dom.querySelectorAll('h2'), h =>
+                h.textContent.trim()
+            )
+        ).toEqual(headers);
+    }
+}
+
+describeMacro('InterfaceOverview', () => {
+    beforeEachMacro(macro => {
+        const realTemplate = macro.ctx.template;
+        macro.ctx.template = jest.fn(async (name, args) => {
+            switch (String(name).toLowerCase()) {
+                case 'interfacedata':
+                    return INTERFACE_DATA;
+                default:
+                    return realTemplate(name, args);
+            }
+        });
```

Comment:
This is a really interesting thing you're doing here... I think this is also worth putting in utils.js, with a name like mockMacro() or defineMacroMock().  I think that it would also need a corresponding unmockMacro() or removeMacroMock() function as well.  

(That doesn't have to happen in this PR if you don't want, however.)

Reply:
I’m doing this in quite a few of my PRs (#1069 comes to mind: https://github.com/mdn/kumascript/blob/4734ac69c79855b0c40abedda3a99b1b82044428/tests/macros/DefaultAPISidebar.test.js#L35-L45